src/main/java/com/github/sergdelft/j2graph/ast/JDTVisitor.java

```diff
@@ -106,7 +106,7 @@ public boolean visit(SimpleName node) {
         // however, it might be that we are not inside a method
         // e.g., field declaration
         // so, we only collect it if we are inside a method
-        if (inAMethod()) {
+        if (inAMethod() && !nonTerminals.get(currentMethod()).isEmpty()) {
```

Comment:
clarification question: why did you have to add this condition everywhere?

I thought the `inAMethod()` was handling it already.

Reply:
`inAMethod()` checks for `methodBuilders` and `nonTerminals` variables. However, if I don't add the check, the call in the next line for `currentNonTerminal()` will call `nonTerminals.get(currentMethod()).peek()` will result in a peek error. So the `nonTerminals` variable is not empty, but the `nonTerminals.get(currentMethod()).peek()` will still fail.

Also, the reason I mentioned earlier that my fixes are a little bit hacked. I added a lot of checks in order to avoid those errors. However, I don't know the actual reason why the peek in `currentNonTerminal()` method will fail. I can probably give you example methods if you would like to debug.

Reply:
If you have one example to send me, that'd be nice. Then, later I can think of the cleanest solution!